.github/workflows/advanced.yml

```diff
@@ -105,29 +105,39 @@ jobs:
       - name: Get workflow SHA
         id: workflow-sha
         run: |
-          echo sha=$(git log -1 --pretty=format:"%H" .github/workflows/ .github/scripts) >> "$GITHUB_OUTPUT"
+          .github/scripts/get-files-sha.sh ${{ github.workspace }} .github/workflows/advanced.yml
 
       - name: Get binutils SHA
         id: binutils-sha
         run: |
-          .github/scripts/get-sha.sh ${{ env.BINUTILS_REPO }} ${{ env.BINUTILS_BRANCH }}
+          .github/scripts/get-repository-sha.sh ${{ env.BINUTILS_REPO }} ${{ env.BINUTILS_BRANCH }}
 
       - name: Get GCC SHA
         id: gcc-sha
         run: |
-          .github/scripts/get-sha.sh ${{ env.GCC_REPO }} ${{ env.GCC_BRANCH }}
+          .github/scripts/get-repository-sha.sh ${{ env.GCC_REPO }} ${{ env.GCC_BRANCH }}
 
       - name: Get MinGW SHA
         id: mingw-sha
         run: |
-          .github/scripts/get-sha.sh ${{ env.MINGW_REPO }} ${{ env.MINGW_BRANCH }}
+          .github/scripts/get-repository-sha.sh ${{ env.MINGW_REPO }} ${{ env.MINGW_BRANCH }}
+
+      - name: Get binutils scripts SHA
+        id: binutils-scripts-sha
+        run: |
+          .github/scripts/get-files-sha.sh ${{ github.workspace }} .github/scripts/binutils
+  
+      - name: Get toolchain scripts SHA
+        id: toolchain-scripts-sha
+        run: |
+          .github/scripts/get-files-sha.sh ${{ github.workspace }} .github/scripts/toolchain
 
       - name: Cache toolchain
         id: cache-toolchain
         uses: actions/cache@v3
         with:
           path: ${{ env.ARTIFACT_PATH }}/${{ env.TOOLCHAIN_PACKAGE_NAME }}
-          key: ${{ env.TOOLCHAIN_NAME }}-${{ steps.workflow-sha.outputs.sha }}-${{ steps.binutils-sha.outputs.sha }}-${{ steps.gcc-sha.outputs.sha }}-${{ steps.mingw-sha.outputs.sha }}
+          key: ${{ env.TOOLCHAIN_NAME }}-${{ steps.workflow-sha.outputs.sha }}-${{ steps.binutils-sha.outputs.sha }}-${{ steps.gcc-sha.outputs.sha }}-${{ steps.mingw-sha.outputs.sha }}-${{ steps.binutils-scripts-sha.outputs.sha }}-${{ steps.toolchain-scripts-sha.outputs.sha }}
```

Comment:
maybe it makes sense to group binutils, gcc and mingw hashes. As the next step of improvements  we would like to create cache for binutils separately.